include/revng/ADT/UpcastablePointer.h

```diff
@@ -212,7 +193,9 @@ class UpcastablePointer {
     return Pointer == P;
   }
 
-  bool operator==(const UpcastablePointer &Other) const {
+  template<UpcastablePointerLike PointerLike>
+    requires(std::equality_comparable_with<T, Dereferenced<PointerLike>>)
+  bool operator==(const PointerLike &Other) const {
```

Comment:
This is to enable comparison with `T *` and `unique_ptr<T>`, correct?

Reply:
Yep, mostly with `T *`.